hcn/hcn.go

```diff
@@ -101,6 +101,20 @@ type HostComputeQuery struct {
 	Filter        string                `json:",omitempty"`
 }
 
+type ExtraParams struct {
+	Resources        interface{} `json:",omitempty"`
+	SharedContainers interface{} `json:",omitempty"`
+	LayeredOn        string      `json:",omitempty"`
+	SwitchGuid       string      `json:",omitempty"`
+	UtilityVM        string      `json:",omitempty"`
+	VirtualMachine   string      `json:",omitempty"`
```

Comment:
What's the difference between these two?

Reply:
No clue at all, I'll have to ask the HNS folks. The only thing this is here for is the `LayeredOn` field, but it felt weird just having that defined instead of the whole thing.

Reply:
Did you find out the answer here?

Reply:
From Jocelyn: "I don't know if they are used. The ExtraParams (and everything under the "Health" struct) contain information about the HNS internals". According to Betsy she thinks the VirtualMachine field may be a pnp instance GUID.

Reply:
So no set answer really